scripts/helpers/getAuthorData.js

```diff
@@ -15,28 +49,15 @@ export const getAuthorData = async () => {
     const authorData = await Promise.all(
       authorNames.map(async (authorName) => {
         try {
-          const resp = await fetch(`/authors/${authorName}.plain.html`);
-      
-          if (!resp.ok) {
-            console.warn(`We could not fetch author data for ${authorName} — Status ${resp.status}`);
-          };
-
-          const dataHTML = await resp.text();
-          const dataContainer = document.createElement('div');
-          dataContainer.innerHTML = dataHTML;
-        
-          return {
-            name: dataContainer.querySelector('h1')?.innerText,
-            title: dataContainer.querySelector('h2')?.innerText,
-            // the HTML picture element
-            image: dataContainer.querySelector('picture') || undefined,
-            // support for multiple paragraphs
-            // sliced due to the image originally contained within a `p`
-            description: [...dataContainer.querySelectorAll('p')]?.slice(1),
-          };
+          return await fetchAuthorData(`/authors/${authorName}.plain.html`);
         } catch (err) {
-          console.error(`We couldn't retreieve author data for ${authorName} - Error ${err}`);
-          return;
+          console.warn(`We couldn't retreieve author data for ${authorName}. Displaying fallback author. - Error ${err}`);
+          try {
+            return await fetchAuthorData(FALLBACK_AUTHOR_URL);
+          } catch (fallbackErr) {
+            console.warn(`We couldn't retreieve fallback author data. - Error ${fallbackErr}`);
+            return FALLBACK_AUTHOR_DATA;
+          };
```

Comment:
The nested try/catches in this function are giving me a little heartburn. What do you think about pulling out the try/catch for the fallback author into it's own function, like `getFallbackAuthor` instead?